library/network/src/modules/SuSEFirewallServices.rb

```diff
@@ -745,6 +780,157 @@ def GetPossiblyConflictServices
+    # @note service_details method.
+    # @return [Boolean] if successful
+    def ReadServicesDefinedByRPMPackages
+      log.info "Reading FirewallD services from #{SERVICES_DIR.join(" and ")}"
+
+      @services ||= {}
+
+      SuSEFirewall.api.services.each do |service_name|
+        # Init everything
+        @services[service_name] = {}
+        @known_services_features.merge!(@known_metadata).each do |_, v|
+          # Set a good name for our service until we read its information
+          case v
+          when "description"
+            # We intentionally don't call the API here. We will use it as a
+            # flag to populate the full service details later on.
+            @services[service_name][v] = default_service_description(service_name)
+          when "name"
+            # We have to call the API here because there are callers which
+            # expect to at least provide a sensible service name without
+            # worrying for the full service details. This is going to be
+            # expensive though since the cost of calling --get-short grows
+            # linearly with the number of available services :-(
+            @services[service_name][v] = SuSEFirewall.api.service_short(service_name)
+          else
+            @services[service_name][v] = []
+          end
+        end
+      end
+    end
+
+    # Returns service definition.
+    # See @services for the format.
+    # If `silent` is not defined or set to `true`, function throws an exception
+    # SuSEFirewalServiceNotFound if service is not found on disk.
+    #
+    # @note Since we do not do full service population in ReadServicesDefinedByRPMPackages
+    # @note we have to do it here but only if the service hasn't been populated
+    # @note before. The way we determine if the service has been populated or not
+    # @note is to look at the "description" key.
+    #
+    # @param [String] service name
+    # @param [String] (optional) whether to silently return nil
+    #                 when service is not found (default false)
+    def service_details(service_name, silent = false)
+      # Drop service: if needed
+      service_name = service_name.partition(":")[2] if service_name.include?("service:")
+      # If service description is the default one then we know that we haven't read the service
+      # information just yet. Lets do it now
+      populate_service(service_name) if all_services[service_name]["description"] ==
+          default_service_description(service_name)
+      service = all_services[service_name]
+      if service.nil? && !silent
+        log.error "Uknown service '#{service_name}'"
+        log.info "Known services: #{all_services.keys}"
+
+        raise(
+          SuSEFirewalServiceNotFound,
+          _("Service with name '%{service_name}' does not exist") % { service_name: service_name }
+        )
+      end
+
+      service
+    end
+
+    # Sets that configuration was modified
+    def SetModified
```

Comment:
Right that's an old artifact where you didn't need to restart the firewall when you modified a service. But now you do. I will fix it.